testing/integration-tests/src/full_client/blocks/mod.rs

```diff
@@ -227,3 +229,61 @@ async fn fetch_block_and_decode_extrinsic_details() {
     assert_eq!(ext.value, 10_000);
     assert!(tx.is_signed());
 }
+
+#[tokio::test]
+async fn decode_signed_extensions_from_blocks() {
```

Comment:
Would be cool to test with different runtimes so both `ChargeAssetTxPayment` and `ChargeTransactionPayment` works but probably annoying to support here.

Could be done in another PR

Reply:
Yeah; I pondered this too but not sure how I'd test this?! It'd be a pain introducing the ability to spawn up a different node just for this

Reply:
I guess one way to unit test this would be crafting some custom metadata with whatever signed extensions we want, and then creating an `OfflineClient` (note the _offline_), setting the metadata on it to be what we want, and then building an extrinsic (could be dynamically constructed). Then, one can manually make the bytes into an `ExtrinsicDetails` and try to decode each of the signed extensions from it.

This way, we could test a few different combinations of signed extensions (eg if Nonce/Tip ones don't exist, or if both the tip ones exist etc).